src/main/java/com/datatorrent/netlet/util/CircularBuffer.java

```diff
@@ -52,13 +56,14 @@
   @SuppressWarnings("unchecked")
   public CircularBuffer(int n, int spin)
   {
-    int i = 1;
-    while (i < n) {
-      i <<= 1;
-    }
+    if (n < 0 || n > MAX_SIZE) throw new RuntimeException(String.format("Error: value too large: %d%n", n));
```

Comment:
%n is the (portable) Java-ism for a newline; I use it to separate the message from the rest of the text.